webapp/components/Firebase/FirebaseInit.ts

```diff
@@ -0,0 +1,53 @@
+// Import the functions you need from the SDKs you need
+import { initializeApp } from 'firebase/app';
+import {
+	getAuth,
+	signInWithPopup,
+	GoogleAuthProvider,
+	Auth,
+	AuthProvider,
+} from 'firebase/auth';
+// TODO: Add SDKs for Firebase products that you want to use
+// https://firebase.google.com/docs/web/setup#available-libraries
+
+// Your web app's Firebase configuration
+// For Firebase JS SDK v7.20.0 and later, measurementId is optional
+const firebaseConfig = {
+	apiKey: process.env.NEXT_PUBLIC_APIKEY,
+	authDomain: process.env.NEXT_PUBLIC_AUTHDOMAIN,
+	projectId: process.env.NEXT_PUBLIC_PROJECTID,
+	storageBucket: process.env.NEXT_PUBLIC_STORAGEBUCKET,
+	messagingSenderId: process.env.NEXT_PUBLIC_MESSAGINGSENDERID,
+	appId: process.env.NEXT_PUBLIC_APPID,
+	measurementId: process.env.NEXT_PUBLIC_MEASUREMENTID,
+};
+
+// Initialize Firebase
+const app = initializeApp(firebaseConfig);
+
+// Initialize Firebase Authentication and get a reference to the service
+const auth = getAuth(app);
+
+const provider = new GoogleAuthProvider();
+
+export const googleSignIn = () => {
```

Comment:
we should handle what happens after you authenticate with Google. Like showing a different UI, saving user basic info to DB, or making user info globally accessible.